scripts/init.sh

```diff
@@ -77,55 +77,63 @@ if [ -r $DEFAULT ]; then
     source $DEFAULT
 fi
 
-function pidofproc() {
-    if [ $# -ne 3 ]; then
-        echo "Expected three arguments, e.g. $0 -p pidfile daemon-name"
-    fi
+if ! typeset -F pidofproc &>/dev/null; then
+    function pidofproc() {
+        if [ $# -ne 3 ]; then
+            echo "Expected three arguments, e.g. $0 -p pidfile daemon-name"
+        fi
 
-    PID=`pgrep -f $3`
-    local PIDFILE=`cat $2`
+        pid=`pgrep -f "^$3"`
```

Comment:
Won't this pick up `influxdabc`? Unlikely I know, but again should be anchored on the right it seems?